apigw-lambda-cognito-sam-java/README.md

```diff
@@ -0,0 +1,125 @@
+   ```bash
+   git clone https://github.com/aws-samples/serverless-patterns
+   ```
+
+2. Change directory to the pattern directory:
+   ```bash
+   cd serverless-patterns/apigw-lambda-cognito-sam-java
+   ```
+
+3. From the command line, execute the below command to build the Java based AWS Lambda function.
+   ```bash
+   sam build
+   ```
+
+4. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```bash
+   sam deploy --guided
+   ```
+4. During the prompts:
+
+   - Enter a stack name
+   - Enter the desired AWS Region.
+   - Allow SAM CLI to create IAM roles with the required permissions.
+   - MyCreateUserFunction has no authentication. Is this okay? [y/N]: select `y`
+   - MyConfirmUserFunction has no authentication. Is this okay? [y/N]: select `y`
+   - MyLoginUserFunction has no authentication. Is this okay? [y/N]: select `y`
+   
+
+   For production applications, you should [enable authentication for the API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-to-api.html) using one of several available options and [follow the API Gateway security best practices](https://docs.aws.amazon.com/apigateway/latest/developerguide/security-best-practices.html).
+   
+   - Keep default values to the rest of the parameters.
+
+   Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+5. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for next step as well as testing.
+
+## How it works
+
+Please refer to the architecture diagram below:
+
+![End to End Architecture](images/architecture.png)
+
+Here's a breakdown of the steps:
+
+1. **Amazon API Gateway**: Receives the HTTP POST request containing the request details in JSON format. There are 3 POST methods exposed for create user, confirm user and login user operations.
+
+2. **AWS Lambda**: Triggered by the API Gateway, this function written in Java uses Amazon Cognito API to create, confirm and login user.
+
+3. **Amazon Cognito**: Depending on the operation invoked, user is created, confirmed or logged in the Amazon Cognito user pool.
+
+4. **Response**: The Lambda function processes the Amazon Cognito API response and sends it back to the user via the API Gateway.
+
+## Testing
+
+1. You can use [curl](https://curl.se/) or any other tool of your choice to send a HTTP POST create user request to the API. Make sure to replace `Endpoint` with the corresponding values from your `sam deploy --guided` output. Also replace all other values. The `email` should be a valid email id.
+
+   ```bash
+   curl -d '{"firstName": "{YourFirstName}", "lastName": "{YourLastName}", "email": "{your-email@mail.com}", "password": "{Y0urP#ssword}"}' -H 'Content-Type: application/json' {CreateUserAPIGatewayEndpoin}
```

Comment:
Please leave the password value empty and rather suggest to use a secure value

Reply:
Fixed